pkg/util/compat/json_slice_merge_test.go

```diff
@@ -0,0 +1,47 @@
+package compat_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	. "github.com/mongodb/mongodb-atlas-kubernetes/pkg/util/compat"
+)
+
+func TestJSONSliceMerge(t *testing.T) {
```

Comment:
Great to have the unit test!
Do you think we can add the use cases:
* when the src has smaller size than the dst?
* the src is empty/nil
* the dst is empty/nil